TELCODOCS-1887: Adding IBI section for standalone deployments only (#82877)
openshift/openshift-docs: modules/cnf-image-based-upgrade-shared-container-partition.adoc

// Module included in the following assemblies:
// * edge_computing/image-based-upgrade/cnf-preparing-for-image-based-upgrade.adoc

ifeval::["{context}" == "ibi-preparing-image-based-install"]
:ibi:
endif::[]

ifeval::["{context}" == "shared-container-partition"]
:ibu:
endif::[]

:_mod-docs-content-type: PROCEDURE
[id="cnf-image-based-upgrade-shared-container-partition_{context}"]
= Configuring a shared container partition between ostree stateroots

ifdef::ibu[]
Apply a `MachineConfig` to both the seed and the target clusters during installation time to create a separate partition and share the `/var/lib/containers` partition between the two `ostree` stateroots that will be used during the upgrade process.

[IMPORTANT]
====
You must complete this procedure at installation time.
====
endif::[]

ifdef::ibi[]

> Apply a `MachineConfig` to the seed cluster to create a separate partition and share the `/var/lib/containers` partition between the two `ostree` stateroots that will be used during the preinstall process.

Reviewer: AFAIR this needs to be done during deployment of the seed, i.e. it's not possible to change this after the cluster is deployed.

Author: There is a note in the comparative IBU [section](https://82877--ocpdocs-pr.netlify.app/openshift-enterprise/latest/edge_computing/image_based_upgrade/preparing_for_image_based_upgrade/cnf-image-based-upgrade-shared-container-partition#cnf-image-based-upgrade-shared-container-partition_shared-container-partition) that says "You must complete this procedure at install time." If I added this note here too, would this help address your comment?

Reviewer: Yep.

Author: Updated with the note, thanks!

----
<1> Specify the root disk.
<2> Specify the start of the partition in MiB. If the value is too small, the installation will fail.

> <3> Specify a minimum size for the partition of 500 GB to ensure adequate disk space for precached images. If the value is too small, the deployments after installation will fail.

Reviewer 2: ```suggestion
<3> Specify a minimum size of 500 GB for the partition. This ensures that adequate disk space for precached images exists.
```

Author: Leaving as is as it effects IBU content too and there was much back and forth about this.
